### ISSUE TITLE
[BUG]  Logitech Saitek Multi Panel AP not Working properly

### ISSUE THREAD

**Issue by Nedo93** [flybywiresim/a32nx#2990]
**Mod Version**
"built": "2021-01-11T06:11:34+00:00",

**Describe the bug**
Trying to do a cold and dark start on FlyByWire A32NX, the Saitek Autopilot multi panel for autopilot, is not working.
It seems like it misses the avionic switch ON for the input. On every other aircraft, the panel is turning on when activating the avionic switch, but here on A32NX mod, I can’t find anything like that.
Thank you in advance for your work.

**To Reproduce**
1. Start a cold and dark flight with a320nx.
2. Execute all checklist and get ready to take off
3. Logitech Saitek multi panel AP is not detecting the avionics ON and I can't use it to setup the AP.

**Expected behavior**
Saitek Multi Panel AP should turn have its display turned on, and you should be able to setup the autopilot functions.

**Actual behavior**
Saitek Multi Panel AP display is blank and AP functions are not working. I can still use the flaps switch.

**Additional context**
I sincerely don't know if this was working before, because I never tried on previous mod versions.

**Comment by phildeg31** [comment]
Hi,
I use the logitech/Saitek Autopilot multi panel and it works fine with the latest FBW version of the mod (because I use also the Thrustmaster TCA Airbus quadrant).

I use SPAD.next that replace the Logitech drivers. It permits to adapt the behaviour of the panel as I want (I did a personal profile).

I do not understand what is your problem.

**Comment by Nedo93** [comment]
I'm not using SPAD.next , my problem is that if I try to do a cold and dark startup, the panel won't switch on. 
If I try to start the flight from the runaway, then the panel works.
In every other plane the panel I can turn on the panel by turning ON the avionic switch, but here on the A32NX, I can't find something like that, and the panel doesn't turn on.

**Comment by Vangthor** [comment]
Same just got my multi-panel and set it up, doesn't seem to react when starting cold and dark. The trim and pitch and auto-throttle work, but the AP doesn't.

**Comment by Benjozork** [comment]
I wonder if it want the avionics bus to be on, which the A320 doesn't really interact with.

Those devices are proprietary and hard to debug. This might take some time to figure out.

**Comment by Vangthor** [comment]
Testing it starting straight on the runway the panel seems to react but I get a "NAV TCAS FAULT" and the everything kinda freaks out haha

**Comment by Nedo93** [comment]
> I wonder if it want the avionics bus to be on, which the A320 doesn't really interact with.
> 
> Those devices are proprietary and hard to debug. This might take some time to figure out.

Yes maybe it's because of this. Probably the panel drivers are hard binded with an avionic switch bus and the A320 doesn't have something like that. I wonder if there is the possibility to intercept the event which turns on the panel and bind it to some button.

**Comment by Vangthor** [comment]
So, a workaround that seems to work for me! Starting cold and dark, do your normal start up procedure up to the point were your engines are up and running and your essentially ready to taxi and take off, then press whatever you have bound to "Auto start engine" which I believe is CRTL+E by default, and nothing in the cabin should change since you're already up and running, a few seconds later the multipanel should come on, works for me at least, perhaps whatever is missing is activated automatically with the auto engine start.

**Comment by phildeg31** [comment]
With SPAD.next all seems fine. I can use it by starting flight on runway or by doing a startup from a gate.

I have this by starting flight on runway directly:
![2](https://user-images.githubusercontent.com/74733393/104233305-f4bc5700-5451-11eb-8e2a-6cf8afde47e9.JPG)

I have this with startup from a gate:
- First, no power for plane so no information on multi panel (virtual power is off in SPAD.next)
![4](https://user-images.githubusercontent.com/74733393/104233355-0a318100-5452-11eb-9c11-f0b9cee389ee.JPG)
- When the autopilot panel has power in plane, multi panel does not detect it yet. I have to force power by activating virtual power in SPAD.next (after that, it works):
![5](https://user-images.githubusercontent.com/74733393/104233370-0e5d9e80-5452-11eb-988c-e60d3f866d45.JPG)

So maybe with SPAD.next you can solve your problem but it does not seem to be a mod issue because I can use multi panel with mod by using SPAD.next.

**Comment by Vangthor** [comment]
Pretty sure you have to buy that software, getting everything up and running and using the CTRL+E to activate whatever is missing to get the panel up and running might be a better option, if it works for you guys as it does for me!

**Comment by Nedo93** [comment]
> Pretty sure you have to buy that software, getting everything up and running and using the CTRL+E to activate whatever is missing to get the panel up and running might be a better option, if it works for you guys as it does for me!

Yes that software is a payware and I don't want to buy something else just to make a 100€ + panel work :D
This soultion seems genius! I will try that later!

**Comment by phildeg31** [comment]
SPAD.next license is 25 euros to use it with FS2020.

**Comment by Vangthor** [comment]
> > Pretty sure you have to buy that software, getting everything up and running and using the CTRL+E to activate whatever is missing to get the panel up and running might be a better option, if it works for you guys as it does for me!
> 
> Yes that software is a payware and I don't want to buy something else just to make a 100€ + panel work :D
> This soultion seems genius! I will try that later!

I hope it works for you bud! Good luck!

**Comment by Vangthor** [comment]
> SPAD.next license is 25 euros to use it with FS2020 but it is better in my opinion. You can configure everithing and to change the panel behaviour as you want.

That's true! Although spending that money JUST to fix the panel from starting seems like a stretch, but of course the software has extended uses that make it worth it, if your actually gonna use all the functions!

**Comment by phildeg31** [comment]
> 
> 
> > SPAD.next license is 25 euros to use it with FS2020 but it is better in my opinion. You can configure everithing and to change the panel behaviour as you want.
> 
> That's true! Although spending that money JUST to fix the panel from starting seems like a stretch, but of course the software has extended uses that make it worth it, if your actually gonna use all the functions!

Yes I did a personal configuration to have a behaviour more closer to real one in plane with SPAD.next and everithing is fine.

In my opinion it is not a bug due to the A320NX mod.

**Comment by Vangthor** [comment]
> > > SPAD.next license is 25 euros to use it with FS2020 but it is better in my opinion. You can configure everithing and to change the panel behaviour as you want.
> > 
> > 
> > That's true! Although spending that money JUST to fix the panel from starting seems like a stretch, but of course the software has extended uses that make it worth it, if your actually gonna use all the functions!
> 
> Yes I did a personal configuration to have a behaviour more closer to real one in plane with SPAD.next and everithing is fine.
> 
> In my opinion it is not a bug due to the A320NX mod.

Yea, I feel it's more MSFS2020 issue!

**Comment by Benjozork** [comment]
Hopefully I can give some context as a developer.

Third party panels will always have problems with third party airplanes, because MSFS is based on a simulation engine that's soon to be 20 years old and has never really been created with flexibility in mind. The framework for integrating additional systems and allowing peripherals to map them out is non-existent.

That's the reason people recommend third party software like SPAD.neXt, because it's really the only way to get an acceptable experience with that hardware.

MS will need to rethink capabilities of third party integrations for this to ever be possible out of the box.

**Comment by phildeg31** [comment]
> 
> 
> > > > SPAD.next license is 25 euros to use it with FS2020 but it is better in my opinion. You can configure everithing and to change the panel behaviour as you want.
> > > 
> > > 
> > > That's true! Although spending that money JUST to fix the panel from starting seems like a stretch, but of course the software has extended uses that make it worth it, if your actually gonna use all the functions!
> > 
> > 
> > Yes I did a personal configuration to have a behaviour more closer to real one in plane with SPAD.next and everithing is fine.
> > In my opinion it is not a bug due to the A320NX mod.
> 
> Yea, I feel it's more MSFS2020 issue!

Or an issue with Logitech drivers which are limited because it works with SPAD.next (it replaces the Logitech drivers).

**Comment by Nedo93** [comment]
> > > Pretty sure you have to buy that software, getting everything up and running and using the CTRL+E to activate whatever is missing to get the panel up and running might be a better option, if it works for you guys as it does for me!
> > 
> > 
> > Yes that software is a payware and I don't want to buy something else just to make a 100€ + panel work :D
> > This soultion seems genius! I will try that later!
> 
> I hope it works for you bud! Good luck!

It worked! You made my day sir.

**Comment by DecMoonFR** [comment]
Another without ctrl+e : add shortcuts (keyboard or button : i mapped fire tests on TCA) on Activate avionic interrupt 1 (& 2) (not sure about english param, in french it's "Activer interrupt géné avionnique ")

**Comment by bluedoors** [comment]
@phildeg31 I'll grab that SPAD.neXt config if i can. I'm also hoping it can correct #1211 where the comms panel doesn't work.

**Comment by alex853** [comment]
Hi!

I have configured shortcut for AVIONICS MASTER 1 ON. This shortcut makes two Logitech Panels operating even after loading of sim on parking (cold&dark). Everything works on Multi Panel. COM1/COM2, DME, XPDR modes work as expected on Radio Panel. I have not tested NAV1/NAV2 and ADF modes.
![IMG_20210620_175639](https://user-images.githubusercontent.com/11858152/122677055-a033b980-d1f1-11eb-972f-59b23c9cdbbb.jpg)

Stable 0.6.1 FBW A320NX + default Logitech software.

**Comment by NemoThunderbolt** [comment]
> Still doesn't work

Me too. Stable or dev. version. I try to use button for AVIONICS MASTER 1 ON and my radio panel does work, but for 1 sec. Then it's off again. It works fine on cessna or if I start A320 from a runway, not cold&dark.
